packages/ckeditor5-engine/src/view/matcher.ts

```diff
@@ -389,27 +386,34 @@ function matchPatterns(
  */
 function normalizePatterns( patterns: PropertyPatterns ): Array<[ true | string | RegExp, true | string | RegExp ]> {
 	if ( Array.isArray( patterns ) ) {
-		return patterns.map( ( pattern: any ) => {
-			if ( isPlainObject( pattern ) ) {
-				if ( pattern.key === undefined || pattern.value === undefined ) {
-					// Documented at the end of matcher.js.
-					logWarning( 'matcher-pattern-missing-key-or-value', pattern );
-				}
+		return patterns.map( pattern => {
+			if ( typeof pattern !== 'object' || pattern instanceof RegExp ) {
+				return [ pattern, true ];
+			}
 
-				return [ pattern.key, pattern.value ];
+			if ( pattern.key === undefined || pattern.value === undefined ) {
+				// Documented at the end of matcher.js.
+				logWarning( 'matcher-pattern-missing-key-or-value', pattern );
 			}
 
-			// Assume the pattern is either String or RegExp.
-			return [ pattern, true ];
+			return [ pattern.key, pattern.value ];
 		} );
 	}
 
-	if ( isPlainObject( patterns ) ) {
-		return Object.entries( patterns );
+	if ( typeof patterns !== 'object' || patterns instanceof RegExp ) {
+		return [ [ patterns as any, true ] ];
+	}
+
+	// Below we do what Object.entries() does, but faster
```

Comment:
```suggestion
	// Below we do what Object.entries() does, but faster.
```

Reply:
Again, surprised that writing something by hand is faster.